src/librustdoc/clean/types.rs

```diff
@@ -799,24 +800,26 @@ impl AttributesExt for [ast::Attribute] {
             Some(item)
         }
 
-        let mut cfg = if doc_cfg_active {
+        let mut cfg = if doc_cfg_active || doc_auto_cfg_active {
             let mut doc_cfg = self
```

Comment:
Do you mind moving this into the first `if` block to enforce that it only has an effect when the feature is enabled?

Reply:
Please also add a test for that (adding doc(cfg) and doc_auto_cfg, but not doc_cfg).

Reply:
Sure.

Reply:
Actually, moving this into the `if` proved to be rather tricky because I still need an `Option` in the end to ensure that at least one `cfg` was found. So I tried to replace `fold` with `reduce` but then realized it would require the first item to be `Cfg::True`. So I'll simply keep the code as is.

Reply:
👍 Seems fine as long as there's a test :)